### PR TITLE
Use short git tag for version and images

### PR DESCRIPTION
**Pull request by hakman**
Fixes: #339

**Does this PR introduce a user-facing change?**:

```
NONE
```

### CODE REVIEW

File: cloudbuild.yaml [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
     - --tag=gcr.io/$PROJECT_ID/cloud-controller-manager:latest
-    - --build-arg=VERSION=$(echo $_GIT_TAG | cut -d - -f2-)
+    - --build-arg=VERSION=$_SHORT_TAG
     - --output=type=registry
     - --platform=linux/amd64,linux/arm64
     - .
 substitutions:
-  _GIT_TAG: '12345'
+  _GIT_TAG: 'v99999999-v12345'
+  _SHORT_TAG: '${_GIT_TAG:10}'
```
Comment (rifelpet) [inline cloudbuild.yaml:19]: for reference: https://cloud.google.com/build/docs/configuring-builds/use-bash-and-bindings-in-substitutions#bash_parameter_expansions

**Review by rifelpet — APPROVED** [review]
this should fix the image builds broken by https://github.com/kubernetes/cloud-provider-aws/pull/350

https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/cloud-provider-aws-push-images/1523933803632922624

/lgtm

File: cloudbuild.yaml (hunk 1 above)
Comment (justinsb) [reply cloudbuild.yaml:19]: Is it possible to have a comment here saying something like `# Remove date prefix (first 10 characters) to create valid semver version, e.g. v20220510-v1.24.0-alpha.0-15-g09bd268 => v1.24.0-alpha.0-15-g09bd268`

**Comment by justinsb** [comment]
I'm going to approve because this feels consistent with #350 and is breaking builds.

cc @nckturner 

/approve
/lgtm

**Pull request merged**